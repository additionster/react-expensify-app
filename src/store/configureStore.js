import { configureStore, combineReducers } from "@reduxjs/toolkit";
import expenseReducer from "../reducers/expenses";
import filterReducer from "../reducers/filters";

export default () => {
    const store = configureStore({
        reducer: combineReducers({
            expenses: expenseReducer,
            filters: filterReducer
        }),
        middleware: (getDefaultMiddleware) => getDefaultMiddleware({
            serializableCheck: false
        }),
        devTools: window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    });
    return store;
};
