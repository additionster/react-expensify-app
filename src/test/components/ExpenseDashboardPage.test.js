import React from "react";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import ExpenseDashboardPage from '../../components/ExpenseDashboardPage';
import { BrowserRouter } from "react-router";
import { Provider } from "react-redux";
import configureStore from "../../store/configureStore";

test('shouldRenderExpenseDashboardPageCorrectly', () => {
    const store = configureStore();
    const renderer = render(
        <BrowserRouter>
            <Provider store={store}>
                <ExpenseDashboardPage />
            </Provider>
        </BrowserRouter>
    );
    expect(renderer.baseElement).toMatchSnapshot();
});