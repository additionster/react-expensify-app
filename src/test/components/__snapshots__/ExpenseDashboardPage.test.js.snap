// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`shouldRenderExpenseDashboardPageCorrectly 1`] = `
<body>
  <div>
    <div>
      <div>
        <h1>
          Viewing 
          0
           
          expenses
           totalling 
          $0.00
        </h1>
      </div>
      <div>
        <input
          type="text"
          value=""
        />
        <select>
          <option
            selected=""
            value="date"
          >
            Date
          </option>
          <option
            value="amount"
          >
            Amount
          </option>
        </select>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class=""
              placeholder="startDate - endDate"
              type="text"
              value="01/01/1970 - 31/01/1970"
            />
            <button
              aria-label="Close"
              class="react-datepicker__close-icon"
              tabindex="-1"
              type="button"
            />
          </div>
        </div>
      </div>
      <div>
        <h1>
          Expense List
        </h1>
        <p>
          No expenses
        </p>
      </div>
    </div>
  </div>
</body>
`;
